### PR TITLE
Reject invalid comparators for `attribute.status` in FileStore

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/12203?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/12203/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 12203
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

`FileStore.search_traces` doesn't throw when `filter_string` contains  `attribute.status < "OK"`, but it should.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [x] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

**Comment by harupy** [comment]
```
>       _validate_search_traces(
            store, [exp_id], "name IN ('trace_0', 'trace_1')", trace_infos[:2][::-1]
        )

exp_id     = '359010433869461112'
generate_trace_infos = TraceInfos(trace_infos=[TraceInfo(request_id='298eba3a6dec48c9a22467b274ce3e63', experiment_id='359010433869461112', t...0052beae44adadd4b0ed6720b234', '9289e7fa5ad54e7aa5a934cc02caac36'], timestamps=[0, 10, 20, 30, 40, 50, 60, 70, 80, 90])
request_ids = ['298eba3a6dec48c9a22467b274ce3e63', '356eb0348fe848faafc8723c176a9ba4', '99799cfadf424fb990079397da218154', 'e6a0f6df477345[309](https://github.com/mlflow/mlflow/actions/runs/9318361069/job/25650537249?pr=12203#step:12:310)f5a50893df42007', 'a5916550982843c5b05f200a35945678', '1f76d43ecdf44585b8d4ac3459ff825b', ...]
store      = <mlflow.store.tracking.file_store.FileStore object at 0x7f1c7ab4c340>
timestamps = [0, 10, 20, 30, 40, 50, ...]
trace_infos = [TraceInfo(request_id='298eba3a6dec48c9a22467b274ce3e63', experiment_id='359010433869461112', timestamp_ms=0, executio...ytest-0/test_search_traces_filter0/mlruns/359010433869461112/traces/1f76d43ecdf44585b8d4ac3459ff825b/artifacts'}), ...]

...
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'mlflow.utils.search_utils.SearchTraceUtils'>, key_type = 'tag'
comparator = 'IN'

    @classmethod
    def is_tag(cls, key_type, comparator):
        if key_type == cls._TAG_IDENTIFIER:
            if comparator not in cls.VALID_TAG_COMPARATORS:
>               raise MlflowException(
                    f"Invalid comparator '{comparator}' not one of '{cls.VALID_TAG_COMPARATORS}"
                )
E               mlflow.exceptions.MlflowException: Invalid comparator 'IN' not one of '{'ILIKE', '!=', '=', 'LIKE'}

cls        = <class 'mlflow.utils.search_utils.SearchTraceUtils'>
comparator = 'IN'
key_type   = 'tag'
```

**Comment by B-Step62** [comment]
> Do we need to support IN?

Interesting, I think SQL/Databricks raises following the behavior for Run Tags, that don't allow `IN` operation. Not sure why we don't support it for run tags, but I found you added `IN` operator for other attributes (https://github.com/mlflow/mlflow/pull/6945)🙃

For me it's fine as long as they are all consistent.

**Comment by WeichenXu123** [comment]
I think if `IN` is not supported in databricks, we shouldn't support it in OSS, otherwise users migrating from OSS to databricks will complain it.

**Comment by harupy** [comment]
@B-Step62 @WeichenXu123 Thanks for the comments. Let's not support it for now to be consistent with databricks.

**Comment by harupy** [comment]
@B-Step62 @serena-ruan Updated the PR. Could you take a look?

File: tests/store/tracking/test_file_store.py [hunk 1]
```diff
@@ -2970,12 +2956,6 @@ def test_search_traces_filter(generate_trace_infos):
         "trace.status = 'ERROR'",
         trace_infos[2:5][::-1],
     )
-    _validate_search_traces(
-        store,
-        [exp_id],
-        "attributes.status IN ('IN_PROGRESS', 'OK')",
```
Comment (B-Step62) [inline tests/store/tracking/test_file_store.py:2976]: Oh I thought we still support `IN` for `attributes` - we support it for runs - probably a bug?
https://github.com/mlflow/mlflow/blob/master/mlflow/utils/search_utils.py#L141

File: tests/store/tracking/test_file_store.py [hunk 2]
```diff
@@ -2951,15 +2945,7 @@ def test_search_traces_filter(generate_trace_infos):
...
         )
-    _validate_search_traces(
-        store,
-        [exp_id],
-        "status IN ('IN_PROGRESS', 'OK')",
-        (trace_infos[:2] + trace_infos[5:])[::-1],
-    )
-    _validate_search_traces(
-        store, [exp_id], "status NOT IN ('IN_PROGRESS', 'OK')", trace_infos[2:5][::-1]
-    )
```
Comment (harupy) [inline tests/store/tracking/test_file_store.py:2962]: this also needs to be restored

**Review by B-Step62 — APPROVED** [review]
LGTM!!

**Pull request merged**